p2p/stream/types/stream.go

```diff
@@ -89,18 +90,19 @@ func (st *BaseStream) WriteBytes(b []byte) (err error) {
 		err = errors.New("message too long")
 		return
 	}
-	if _, err = st.rw.Write(intToBytes(len(b))); err != nil {
-		return
-	}
-	bytesWriteCounter.Add(sizeBytes)
-	if _, err = st.rw.Write(b); err != nil {
-		return
+	size := sizeBytes + len(b)
+	message := make([]byte, size)
+	copy(message, intToBytes(len(b)))
+	copy(message[sizeBytes:], b)
+
+	if _, err = st.raw.Write(message); err != nil {
```

Comment:
Shall we also add the lock to Write method?

Reply:
Actually, we use write only once and no need for lock here, but libp2p seems so complicated, it's better just to add it and don't worry. Added.